c/meterpreter/source/metsrv/remote_dispatch.c

```diff
@@ -146,14 +147,36 @@ DWORD request_core_enumextcmd(Remote* remote, Packet* packet)
 	return ERROR_SUCCESS;
 }
 
+/*
+ * Deinitialize the core pseudo extension
+ */
+static DWORD deinit_server_extension(Remote* remote)
+{
+	command_deregister_all(customCommands);
+	deregister_base_dispatch_routines();
+
+	return ERROR_SUCCESS;
+}
+
 /*
  * Registers custom command handlers
  */
 VOID register_dispatch_routines()
 {
 	gExtensionList = list_create();
 
+	Command* pFirstCommand = register_base_dispatch_routines();
 	command_register_all(customCommands);
+
+	PEXTENSION pExtension = (PEXTENSION)malloc(sizeof(EXTENSION));
+	if (pExtension) {
+		memset(pExtension, 0, sizeof(EXTENSION));
+		pExtension->deinit = deinit_server_extension;
+		pExtension->end = pFirstCommand;
```

Comment:
The first command in the array should be the last command for the extension because when they are registered in a sort of stack which causes the order to be reversed. Likewise, the `extensionCommand` pointer is the top of the stack and is thus the starting command for the extension.